animations-dbus/animations-dbus-server-animation-manager.c

```diff
@@ -0,0 +1,382 @@
+
+  if (!g_hash_table_lookup_extended (priv->animation_effects,
+                                     GUINT_TO_POINTER (effect_id),
+                                     NULL,
+                                     (gpointer *) &server_animation_effect))
+    {
+      g_set_error (error,
+                   ANIMATIONS_DBUS_ERROR,
+                   ANIMATIONS_DBUS_ERROR_NO_SUCH_EFFECT,
+                   "No such effect with id %u",
+                   effect_id);
+      return NULL;
+    }
+
+  return server_animation_effect;
+}
+
+gboolean
+animations_dbus_server_animation_manager_export (AnimationsDbusServerAnimationManager  *server_animation_manager,
+                                                 const char                            *object_path,
+                                                 GError                               **error)
+{
+  AnimationsDbusServerAnimationManagerPrivate *priv = animations_dbus_server_animation_manager_get_instance_private (server_animation_manager);
+
+  return g_dbus_interface_skeleton_export (G_DBUS_INTERFACE_SKELETON (server_animation_manager),
+                                           priv->connection,
+                                           object_path,
+                                           error);
+}
+
+/**
+ * animations_dbus_server_animation_manager_create_effect:
+ * @server_animation_manager: An #AnimationsDbusServerAnimationManager.
+ * @title: The title of the effect.
+ * @name: The name of the effect to be created.
+ * @settings: A #GVariant of settings forming the initial settings payload.
+ * @error: A #GError.
+ *
+ * Create a #AnimationsDbusServerEffect for use by the
+ * server side. The effect will appear on the bus but
+ * will be managed by the caller.
+ *
+ * Returns: (transfer full): A #AnimationsDbusServerAnimationManager
+ * or %NULL with @error set in case of an error.
+ */
+AnimationsDbusServerEffect *
+animations_dbus_server_animation_manager_create_effect (AnimationsDbusServerAnimationManager  *server_animation_manager,
+                                                        const char                            *title,
+                                                        const char                            *name,
+                                                        GVariant                              *settings,
+                                                        GError                               **error)
+{
+  AnimationsDbusServerAnimationManagerPrivate *priv = animations_dbus_server_animation_manager_get_instance_private (server_animation_manager);
+  g_autoptr(GError) local_error = NULL;
+
+  AnimationsDbusServerEffectBridge *effect_bridge =
+    animations_dbus_server_effect_factory_create_effect (priv->effect_factory,
+                                                         name,
+                                                         settings,
+                                                         error);
```

Comment:
Isn't this leaked?

Reply:
It is, yes - thanks for picking up on that.